src/debug/javaDockerResolver.ts

```diff
@@ -0,0 +1,71 @@
+import * as vscode from 'vscode';
+
+import { IDockerParser, IDockerResolver, PortInfo } from './debugInterfaces';
+import { Kubectl } from '../kubectl';
+import { shell } from '../shell';
+
+const defaultJavaDebugPort = "5005";
+const defaultJavaAppPort = "9000";
+const defaultJavaDebugOpts = `-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=${defaultJavaDebugPort},quiet=y`;
+const javaDebugOptsRegExp = /(-agentlib|-Xrunjdwp):\S*(address=[^\s,]+)/i;
+const fullJavaDebugOptsRegExp = /^java\s+.*(-agentlib|-Xrunjdwp):\S*(address=[^\s,]+)\S*/i;
+
+export class JavaDockerResolver implements IDockerResolver {
+    public isSupportedImage(baseImage: string): boolean {
+        return baseImage.indexOf("java") >= 0
+            || baseImage.indexOf("openjdk") >= 0
+            || baseImage.indexOf("oracle") >= 0;
+    }
+
+    public async resolvePortsFromFile(dockerParser: IDockerParser, env: {}): Promise<PortInfo> {
+        const portInfo: PortInfo = {

```

Comment:
const to me implies this won't be modified, which it clearly will be.  (I know the _actual_ restriction is that the _reference_ won't be modified, so perhaps I'm being silly and other readers won't take away the same assumption.)